implementations/micrometer-registry-influx/src/main/java/io/micrometer/influx/InfluxMeterRegistry.java

```diff
@@ -162,6 +148,88 @@ protected void publish() {
+                    && meterName.startsWith(prefix + ".");
+
+            return prefixes.stream().filter(matchedPredicate).findFirst()
+                    .map(matchedPrefix -> new MeterKey(meterId.getType(), matchedPrefix, meterId.getTags(), baseUnit))
+                    .orElse(null);
+        }
+        return null;
+    }
+
+    private void publishMetrics(String influxEndpoint, List<Meter> meters) throws Throwable {
+        logger.debug("publish meters without prefixes");
+        for (List<Meter> batch : InfluxMeterPartition.partition(meters, config.batchSize())) {
+            publishMetrics(influxEndpoint, batch.size(),
+                    batch.stream()
+                            .flatMap(m -> m.match(gauge -> writeGauge(gauge.getId(), gauge.value()),
+                                    counter -> writeCounter(counter.getId(), counter.count()), this::writeTimer,
+                                    this::writeSummary, this::writeLongTaskTimer,
+                                    gauge -> writeGauge(gauge.getId(), gauge.value(getBaseTimeUnit())),
+                                    counter -> writeCounter(counter.getId(), counter.count()), this::writeFunctionTimer,
+                                    this::writeMeter))
+                            .collect(joining("\n")));
+        }
+    }
+
+    private void publishMetrics(String influxEndpoint, Map<MeterKey, List<Meter>> metersMap) throws Throwable {
+        logger.debug("publish meters with prefixes");
+        for (List<MeterKey> batch : InfluxMeterPartition.partition(new ArrayList<>(metersMap.keySet()),
+                config.batchSize())) {
+            publishMetrics(influxEndpoint, batch.size(),
+                    batch.stream()
+                            .flatMap(k -> writeMetersAsSingleMultiFieldLine(metersMap.get(k),
+                                    meter -> match(meter, Gauge::value, Counter::count,
+                                            timeGauge -> timeGauge.value(getBaseTimeUnit()), FunctionCounter::count),
+                                    getConventionName(k)))
+                            .collect(joining("\n")));
+        }
+    }
+
+    private void publishMetrics(String influxEndpoint, int metersCount, String content) throws Throwable {
+        logger.debug("send to InfluxDB metrics: {}", content);
+
+        HttpSender.Request.Builder requestBuilder = httpClient.post(influxEndpoint)
+                .withBasicAuthentication(config.userName(), config.password());
+
+        config.apiVersion().addHeaderToken(config, requestBuilder);
+
+        requestBuilder.withPlainText(content).compressWhen(config::compressed).send().onSuccess(response -> {
+            logger.debug("successfully sent {} metrics to InfluxDB.", metersCount);
+            databaseExists = true;
+        }).onError(response -> logger.error("failed to send metrics to influx: {}", response.body()));
```

Comment:
I am wondering if maybe this is a time when we can add something more (more details) which help us with further investigation f.e. HTTP status code? (based on our experience) what do u think @eugene-sadovsky ?

Reply:
checked the logs, I think we're good here. We either have stack trace or response body is pretty descriptive.. I'd like to have as little changes as possible